### DISCUSSION TITLE
GET /event equivalent for v4?

### DISCUSSION THREAD

**Discussion by Serkan-devel** [community/community#24817] [Apps, API and Webhooks]
<p>So, I’m trying to switch my small project from REST to GraphQL. Looks nice to fetch exactly what’s needed.</p>
<p>But what I’m mostly fetching from REST are <a href="https://developer.github.com/v3/activity/events/#events" rel="nofollow noopener">events</a> from v3 but i can’t find it on the v4-docs.</p>
<p>Is there any equivalent to events on GraphQL?</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/serkan-devel">@serkan-devel</a>,</p>
<p>Thank you for being here! In GraphQL there is an issue timeline connection, which looks like it should work for you. <a href="https://developer.github.com/v4/union/issuetimelineitem/" rel="nofollow noopener">https://developer.github.com/v4/union/issuetimelineitem/</a></p>
<p>I hope this helps!</p>
<p>Andrea</p>

**Comment by Serkan-devel** [comment]
<p>Thank you for the answer. I just succeeded with a working prototype of the app in REST just recently, but using graphql would probably make the code shorter, going through less parsing steps.</p>
<p>But for some reason, I can’t form calls, like this one</p>
<pre><code>query {
  organization(login:"github") {
    IssueTimelineItems(first:20) {
      edges {
        node {
          id
          title
          url
        }
      }
    }
  }
}
</code></pre>
<p>How does on form IssueTimelineItems within a call?</p>

**Comment by brunolemos** [comment]
<p>IssueTimeline is not a replacement for the /events v3 API because it seems to be per issue only.</p>
<p>For example, I need all events from the repository, from all issues, all pull requests, branches created, starred events, etc. Just like the api v3 works. Also user received_events, organization events, notifications, etc.</p>
<p>Is that planned?</p>
<p>I believe this should not be marked as solved.</p>

**Reply by alanhamlett** [reply]
Here's a new question for full `/events` support in v4:

https://github.com/orgs/community/discussions/42535

**Comment by Kagemaru** [comment]
<p>We’re in the same spot, trying to port over an application that uses the /events of the V3 API. An equivalent stream would be awesome in the GraphQL API</p>

**Comment by Nosferican** [comment]
<p>For example, today in the GitHub Universe GraphQL workshop by <a href="https://twitter.com/moggytech" rel="nofollow noopener"><strong>@</strong></a><strong><a href="https://twitter.com/krider2010" rel="nofollow noopener">krider2010</a>,</strong> the exercises included</p>
<blockquote>
<p>Which was your first (created) repository on Github?</p>
</blockquote>
<p>GraphQL currently doesn’t support a way to query CreateRepo event for answering that question. That is one use case where one has to fallback to the RESTful API (i.e.,  /users/:username/events endpoint).</p>
<p>As an aside, per the v3 API documentation, you wouldn’t be able to access your first CreateRepo event for your very first repository unless your first ever repository was contained in your last 300 events  / within the 90 days period.</p>

**Comment by fleskesvor** [comment]
<p>I’m in the same spot, trying to query push and release events specifically, but the issue timeline connection seems far too limited for most use cases.</p>
